### PR TITLE
[Iceberg]Fix test failures caused by concurrently handle the same table

### PR DESCRIPTION
**Pull request by hantangwangd**
## Description

The PR fix the test failures in `TestRemoveOrphanFilesProcedureBase`. These failures are caused  by concurrently running the test methods which handle table with the same name in the same catalog, so may affect each other.

## Motivation and Context

Fix test failures appear in: https://github.com/prestodb/presto/actions/runs/10184749910/job/28172882513?pr=23341

## Impact

N/A

## Test Plan

N/A

## Contributor checklist

- [ ] Please make sure your submission complies with our [development](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#development), [formatting](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#formatting), [commit message](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#commit-formatting-and-pull-requests), and [attribution guidelines](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#attribution).
- [ ] PR description addresses the issue accurately and concisely.  If the change is non-trivial, a GitHub Issue is referenced.
- [ ] Documented new properties (with its default value), SQL syntax, functions, or other functionality.
- [ ] If release notes are required, they follow the [release notes guidelines](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines).
- [ ] Adequate tests were added if applicable.
- [ ] CI passed.

## Release Notes

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

File: presto-iceberg/src/test/java/com/facebook/presto/iceberg/procedure/TestRemoveOrphanFilesProcedureHadoop.java [hunk 1]
```diff
@@ -42,10 +43,10 @@ public TestRemoveOrphanFilesProcedureHadoop()
         super(HADOOP, ImmutableMap.of());
     }
 
-    @Override
+    @Test(dataProvider = "timezones")
```
Comment (ZacBlanco) [inline presto-iceberg/src/test/java/com/facebook/presto/iceberg/procedure/TestRemoveOrphanFilesProcedureHadoop.java:46]: is this change intentional?

Comment (hantangwangd) [reply presto-iceberg/src/test/java/com/facebook/presto/iceberg/procedure/TestRemoveOrphanFilesProcedureHadoop.java:46]: Yes, seems we have to make this change after adding `@Test(singleThreaded = true)` on `TestRemoveOrphanFilesProcedureBase`, or we will get an exception as follows when running `TestRemoveOrphanFilesProcedureHadoop`:

```
Cannot inject @Test annotated Method [testRemoveOrphanFilesWithNonDefaultMetadataPath] with [class java.lang.String, boolean].
```

**Review by ZacBlanco — APPROVED** [review]
